gen3/lib/g3k_manifest.sh

```diff
@@ -0,0 +1,117 @@
+#!/bin/bash
+#
+# Little library of manifest related functions usually accessed
+# via the `g3k` cli defined in kube/kubes.sh
+#
+
+G3K_MANIFEST_SH_DIR=$(dirname "${BASH_SOURCE:-$0}")  # $0 supports zsh
+GEN3_HOME="${GEN3_HOME:-$(cd "${G3K_MANIFEST_SH_DIR}/../.." && pwd)}"
+GEN3_MANIFEST_HOME="${GEN3_MANIFEST_HOME:-"$(cd "${GEN3_HOME}/.." && pwd)/cdis-manifest"}"
+export GEN3_HOME
+export GEN3_MANIFEST_HOME
+
+#
+# Internal init helper
+#
+g3k_manifest_init() {
+  # do this at most once a day
+  local doneFilePath="$XDG_RUNTIME_DIR/g3kManifestInit_$(date +%Y%m%d)"
+  if [[ (! "$1" =~ ^-*force$) && -f "${doneFilePath}" ]]; then
+    return 0
+  fi
+  if [[ ! -d "${GEN3_MANIFEST_HOME}" ]]; then
+    echo -e $(red_color "ERROR: GEN3_MANIFEST_HOME does not exist: ${GEN3_MANIFEST_HOME}")
+    echo "git clone https://github.com/uc-cdis/cdis-manifest.git ${GEN3_MANIFEST_HOME}"
+    # This will fail if proxy is not set correctly
+    git clone "https://github.com/uc-cdis/cdis-manifest.git" "${GEN3_MANIFEST_HOME}"
+  fi
+  if [[ -d "$GEN3_MANIFEST_HOME/.git" ]]; then
```

Comment:
how would git fetch work if `.git` doesn't exist?

Reply:
yeah - if the folder is not managed by `git`, then don't try to do a `git fetch`.
this guard only comes into play when we're running the test suite - where we point the code at a folder with some staged test manifests ...

Reply:
oh sorry, read it in the other way